nilearn/plotting/img_plotting.py

```diff
@@ -208,6 +208,11 @@ class _MNI152Template(object):
     vmax   = None
     _shape  = None
 
+    def __init__(self, *args, **kwargs):
+        # This looks like a hack but we want a constant image compatible with
+        # all interfaces
+        pass
```

Comment:
What happens is that this template is added to the plots using `add_overlay`. However, `add_overlay` has an epsilon threshold used to remove numerical noise. This threshold is applied to the image, and thus a new image is created using new_img_like. However, the empty constructor in the template forbids allocation using this code:
`ref_img.__class__(data, affine, header=header)`. This is why I chose a polymorphic constructor to solve this problem.

Several options:
* The polymorphic constructor
* Have a custom type that does not allow modifications
* Use a clever code to fallback to Nifti1Image if a constructor fails in `new_img_like`
* Make the template derive from Nifti1Image and make clever code to find a suitable constructor in super classes in `new_img_like`
* Maybe other options I don't see.